pydantic/utils.py

```diff
@@ -393,10 +399,19 @@ def _normalize_indexes(
 
         >>> self._normalize_indexes({0, -2, -1}, 4)
         {0, 2, 3}
+        >>> self._normalize_indexes({'__all__'}, 4)
+        {0, 1, 2, 3}
         """
         if self._type is set:
+            if '__all__' in items:
+                return {i for i in range(v_length)}
```

Comment:
should it be an error if there's anything else in the set besides `'__all__'`?

Reply:
Hmm, don't have a strong opinion either way.  It certainly doesn't make sense to intentionally put `__all__` along with other indices in the set.  But I don't think the behavior would be surprising or unexpected in this case either.

Reply:
Let's raise an error.